python/lsst/analysis/tools/atools/sourceInjectionPlots.py

```diff
@@ -0,0 +1,286 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = (
+    "CompletenessPurityTool",
+    "TargetInjectedCatDelta",
+    "TargetInjectedCatDeltaScatterAstrom",
+    "TargetInjectedCatDeltaRAScatterPlot",
+    "TargetInjectedCatDeltaDecScatterPlot",
+    "TargetInjectedCatDeltaScatterPhotom",
+    "TargetInjectedCatDeltaPsfScatterPlot",
+    "TargetInjectedCatDeltaCModelScatterPlot",
+    "TargetInjectedCatDeltaMetrics",
+)
+
+from lsst.pex.config import Field
+
+from ..actions.plot.completenessPlot import CompletenessHist
+from ..actions.plot.scatterplotWithTwoHists import ScatterPlotStatsAction, ScatterPlotWithTwoHists
+from ..actions.scalar.scalarActions import MagPercentileAction, MedianAction, SigmaMadAction
+from ..actions.vector import (
+    ConvertFluxToMag,
+    ConvertUnits,
+    DownselectVector,
+    LoadVector,
+    MagDiff,
+    RAcosDec,
+    RangeSelector,
+    SnSelector,
+    SubtractVector,
+)
+from ..contexts import CoaddContext
+from ..interfaces import AnalysisTool
+
+
+class CompletenessPurityTool(AnalysisTool):
+    """Plot the completeness or purity of injected sources by magnitude."""
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        self.process.buildActions.mag = ConvertFluxToMag(vectorKey="ref_{band}_flux")
+        self.process.buildActions.matchDistance = LoadVector(vectorKey="match_distance")
+
+        self.process.calculateActions.mag50 = MagPercentileAction()
+        self.process.calculateActions.mag50.vectorKey = "ref_{band}_flux"
+        self.process.calculateActions.mag50.matchDistanceKey = "match_distance"
+
+        self.process.calculateActions.mag20 = MagPercentileAction(percentile=20.0)
+        self.process.calculateActions.mag20.vectorKey = "ref_{band}_flux"
+        self.process.calculateActions.mag20.matchDistanceKey = "match_distance"
+
+        self.produce.plot = CompletenessHist()
+        self.produce.metric.units = {"mag20": "mag", "mag50": "mag"}
+        self.produce.metric.newNames = {"mag20": "{band}_mag20", "mag50": "{band}_mag50"}
+
+
+class TargetInjectedCatDelta(AnalysisTool):
+    """Plot the difference between a target catalog and an
+    injected catalog for the quantity set in `setDefaults`.
+    """
+
+    parameterizedBand = Field[bool](
+        doc="Does this AnalysisTool support band as a name parameter", default=True
```

Comment:
This doesn't need to be here, just set parametrizeBand = True/False at the top.

Reply:
These are all Trys' original plots using the atools matcher that are no longer being used, sorry for making you read them (again).

Reply:
Are any of my plots getting used? I have them all saved on another PR so you can delete them if they're redundant.

Reply:
Not here, no. I only kept what is still being used from your original commits.